src/registrar/services/cloudflare_service.py

```diff
@@ -0,0 +1,96 @@
+import json
+import logging
+from django.conf import settings
+
+from registrar.utility.errors import APIError
+from registrar.utility.api_helpers import make_api_request
+
+logger = logging.getLogger(__name__)
+
+
+class CloudflareService:
+
+    def __init__(self):
+        self.base_url = "https://api.cloudflare.com/client/v4"
+        self.service_email = settings.SECRET_DNS_SERVICE_EMAIL
+        self.tenant_key = settings.SECRET_DNS_TENANT_KEY
+        self.tenant_id = settings.SECRET_DNS_TENANT_ID
+        self.headers = {
+            "X-Auth-Email": self.service_email,
+            "X-Auth-Key": self.tenant_key,
+            "Content-Type": "application/json",
+        }
+
+    def create_account(self, account_name):
+        url = f"{self.base_url}/accounts"
+        data = {"name": account_name, "type": "enterprise", "unit": {"id": self.tenant_id}}
+        response = make_api_request(url=url, method="POST", headers=self.headers, data=data)
+        if not response["success"]:
+            raise APIError(f"Failed to create account for {account_name}: {response['details']}")
+
+        logger.info(f"Created host account: {response['data']}")
+
+        return response["data"]
+
+    def create_zone(self, zone_name, account_id):
+        url = f"{self.base_url}/zones"
+        data = {"name": zone_name, "account": {"id": account_id}}
+        response = make_api_request(url=url, method="POST", headers=self.headers, data=data)
+        errors = response.get("errors")
+        if not response["success"]:
+
+            raise APIError(
+                f"Failed to create zone for account {account_id}: errors: {errors} message:" +
```

Comment:
```suggestion
                f"Failed to create zone for account {account_id}: errors: {errors} message: " +
```
Noticed `test_create_zone_failure` is looking for a space here!